protocol/communities/manager.go

```diff
@@ -445,15 +445,18 @@ func (m *Manager) Start() error {
 	if m.ownerVerifier != nil {
 		m.runOwnerVerificationLoop()
 	}
+	return nil
+}
 
-	if m.torrentConfig != nil && m.torrentConfig.Enabled {
-		err := m.StartTorrentClient()
-		if err != nil {
-			m.LogStdout("couldn't start torrent client", zap.Error(err))
+func (m *Manager) SetOnline(online bool) {
+	if online {
+		if m.torrentConfig != nil && m.torrentConfig.Enabled && !m.TorrentClientStarted() {
+			err := m.StartTorrentClient()
+			if err != nil {
+				m.LogStdout("couldn't start torrent client", zap.Error(err))
+			}
```

Comment:
Should this also stop the client if `!online` ?

Reply:
Hey @0x-r4bbit, 
yeah, I have thought about stopping the client when `!online`. But I have 2 concerns:

1. `StartTorrentClient` creates a new instance of `TorrentClient`. This means that any previous calls to `torrentClient.AddTorrent` may be lost (e.g. it's called from `SeedHistoryArchiveTorrent`). And the same for `AddMagnet`. Is it okay to rebuild `torrentClient` without restoring torrents/magents?
To be on the safe side I only use the `setOnline` method to delay the creation of the `TorrentClient`. 

2.  There is also an api call `EnableCommunityHistoryArchiveProtocol (and Disable)` that starts/stops the `torrentClient` (which could currently be called from the Desktop Settings). And I thought that the behaviour of `setOnline` and `EnableCommunityHistoryArchiveProtocol` might be inconsistent.

Maybe I'm missing other problems.

Reply:
> There is also an api call EnableCommunityHistoryArchiveProtocol (and Disable) that starts/stops the torrentClient (which could currently be called from the Desktop Settings). And I thought that the behaviour of setOnline and EnableCommunityHistoryArchiveProtocol might be inconsistent.

Yes, as a matter of fact, back then, it was a requirement for the torrent to always be bootstrapped when using desktop clients. The setting to enable/disable the protocol was more of a thing for development (this might have changed by now).

Considering that, it indeed makes sense to keep the client running once it has started (unless the protocol is disabled entirely).
The reason I brought this up here is because I thought, when `!online`, the torrent client might cause high CPU usage again, but maybe that's not the case?

Reply:
> Yes, as a matter of fact, back then, it was a requirement for the torrent to always be bootstrapped when using desktop clients. The setting to enable/disable the protocol was more of a thing for development (this might have changed by now).
> 
> Considering that, it indeed makes sense to keep the client running once it has started (unless the protocol is disabled entirely).

I see, thank you for clarifying that.

> The reason I brought this up here is because I thought, when !online, the torrent client might cause high CPU usage again, but maybe that's not the case?

Yes, the torrent client only uses a lot of CPU when it's started while offline.

Here I'm enabling and disabling the network, but the CPU usage remains low:
https://github.com/status-im/status-go/assets/1083341/fc1bd684-8525-4095-8818-541407646583